Update to use wikit: 2.0.0-alpha.1

Ladsgroup opened the pull request:
- Takes advantage of Dropdown menu
 - Set tag for property lookup search results

Ladsgroup committed:
Update to use wikit: 2.0.0-alpha.1

 - Takes advantage of Dropdown menu
 - Set tag for property lookup search results

micgro42 commented on src/components/PropertyLookup.vue:
```diff
@@ -34,7 +36,13 @@ export default Vue.extend( {
 		async search( newSearchString: string ): Promise<void> {
-			this.searchResults = await this.$store.dispatch( 'searchProperties', newSearchString );
+			const searchResults = await this.$store.dispatch( 'searchProperties', newSearchString );
+			this.searchResults = searchResults.map( ( key: MenuItem & SearchResult ) => {
```
Can we move this into the action?

Ladsgroup replied:
It needs i18n and that's not accessible in outside of the vue object. I can use it but it'll be pretty messy.

micgro42 replied:
Mh, I'll have a look. Having the i18n here, but the assigning the tag in the action should work and already make things a lot nicer

bereket-WMDE commented on src/components/ValueTypeDropDown.vue:
```diff
@@ -57,12 +72,7 @@ export default Vue.extend( {
-			border-width: 1px;
-			border-radius: 2px;
-			appearance: auto;
+			margin-block-start: 21px;
```
shouldn't all this style be removed ? 
AFAIK, the styles are already implemented in the DS. 
just like in the lookup component

Ladsgroup replied:
I removed most of them, but If I remove the rest, it'll look like this:
![image](https://user-images.githubusercontent.com/5351225/101170374-a63fbf80-363e-11eb-936f-c8949e915226.png)

bereket-WMDE replied:
hmm... @micgro42 could you verify that we don't have any style defined for the lookup component in the QB ?
if so, should it have been the same case with the dropdown ?
thanks

micgro42 replied:
will have a look

micgro42 replied:
I moved the responsibility of arranging the child components to the QueryCondition component

micgro42 commented on src/components/ValueTypeDropDown.vue:
```diff
@@ -4,20 +4,12 @@
+		<Dropdown
 			class="querybuilder__dropdown-select"
-			v-model="selected"
+			v-model="selectedItem"
```
No, we can make all that much simpler now. We basically just listen to the `input` event and re-emit a cleaned version that contains only the value we care about. Conversely, we can set the `value` prop here directly.

Then we can also drop all the `watch` methods and the entire local state. We would then only have something like two computed methods and an event listener left

Ladsgroup replied:
I wrestled with it for hours before making this PR, I admit I don't know vue well enough to do what you mean and make it work. Can you try it?

micgro42 replied:
Will do!

micgro42 commented on src/components/ValueTypeDropDown.vue:
```diff
@@ -26,12 +18,19 @@
+				label: PropertyValueRelation.Matching,
+				description: '',
+			},
+			optionItems: Object.values( PropertyValueRelation ).map( ( value: PropertyValueRelation ) => {
```
Those option items should have translations. They could have the form of
```
{ 
  value: PropertyValueRelation.Matching,
  label: i18n('query-builder-property-lookup-' + PropertyValueRelation.Matching ),
}
```
 or something like that

Ladsgroup replied:
I noticed the problem way before the upgrade, I feel this is out of scope of this PR, we should fix it soon but separately

micgro42 replied:
I think that is fair. Could you create a ticket in the current iteration board for this while I look at the other issues?

micgro42 committed:
Simplify piping the selected elements

There will always be one option selected because we have one option for
every valid prop. Unfortunately, TypeScript is not able to infer that
and so this suppresses the `| undefined` return type of
Array.prototype.find.

micgro42 committed:
Declare i18n plugin globally

See for documentation:
https://vuejs.org/v2/guide/typescript.html#Augmenting-Types-for-Use-with-Plugins

micgro42 committed:
Evaluate datatype in action

This keeps the logic in the action and the component has only to deal
with i18n.

micgro42 committed:
Move responsibility for layout to the parent

Not the component themselves should know how they are placed, but they
should be arranged by the parent.

Since the styles are scoped, we can use a classname relative to the
local class as BEM block identifier.

Ladsgroup committed:
Merge branch 'master' into use_new_wikit_yay
